spring-batch-integration/src/test/java/org/springframework/batch/integration/chunk/RemoteChunkingMasterStepBuilderTest.java

```diff
@@ -0,0 +1,264 @@
+
+		this.expectedException.expect(IllegalArgumentException.class);
+		this.expectedException.expectMessage("An OutputChannel must be provided");
+
+		// when
+		TaskletStep step = builder.build();
+
+		// then
+		// expected exception
+	}
+
+	@Test
+	public void testUnsupportedOperationExceptionWhenSpecifyingAnItemWriter() {
+		// given
+		this.expectedException.expect(UnsupportedOperationException.class);
+		this.expectedException.expectMessage("When configuring a master " +
+				"step for remote chunking, the item writer will be automatically " +
+				"set to an instance of ChunkMessageChannelItemWriter. " +
+				"The item writer must not be provided in this case.");
+
+		// when
+		TaskletStep step = new RemoteChunkingMasterStepBuilder<String, String>("step")
+				.reader(this.itemReader)
+				.writer(items -> { })
+				.repository(this.jobRepository)
+				.transactionManager(this.transactionManager)
+				.inputChannel(this.inputChannel)
+				.outputChannel(this.outputChannel)
+				.build();
+
+		// then
+		// expected exception
+	}
+
+	@Test
+	public void testMasterStepCreation() {
+		// when
+		TaskletStep taskletStep = new RemoteChunkingMasterStepBuilder<String, String>("step")
+				.reader(this.itemReader)
+				.repository(this.jobRepository)
+				.transactionManager(this.transactionManager)
+				.inputChannel(this.inputChannel)
+				.outputChannel(this.outputChannel)
+				.build();
+
+		// then
+		Assert.assertNotNull(taskletStep);
+	}
+
+	/*
+	 * The following test is to cover setters that override those from parent builders.
+	 */
+	@Test
+	public void testSetters() {
+		// when
+		TaskletStep taskletStep = new RemoteChunkingMasterStepBuilder<String, String>("step")
+				.reader(this.itemReader)
+				.readerIsTransactionalQueue()
+				.repository(this.jobRepository)
+				.transactionManager(this.transactionManager)
+				.transactionAttribute(new DefaultTransactionAttribute())
+				.inputChannel(this.inputChannel)
+				.outputChannel(this.outputChannel)
+				.listener(new Object())
+				.listener(new SkipListenerSupport<>())
+				.listener(new ChunkListenerSupport())
+				.listener(new StepExecutionListenerSupport())
+				.listener(new CompositeItemReadListener<>())
+				.listener(new CompositeItemWriteListener<>())
+				.listener(new RetryListenerSupport())
+				.skip(Exception.class)
+				.noSkip(RuntimeException.class)
+				.skipLimit(10)
+				.retry(Exception.class)
+				.noRetry(RuntimeException.class)
+				.retryLimit(10)
+				.retryContextCache(new MapRetryContextCache())
+				.noRollback(Exception.class)
+				.chunk(10)
+				.startLimit(3)
+				.allowStartIfComplete(true)
+				.exceptionHandler(new DefaultExceptionHandler())
+				.stepOperations(new RepeatTemplate())
+				.chunkOperations(new RepeatTemplate())
+				.backOffPolicy(new NoBackOffPolicy())
+				.stream(new ItemStreamSupport() {})
+				.keyGenerator(Object::hashCode)
+				.build();
+
+		// then
+		Assert.assertNotNull(taskletStep);
```

Comment:
Just because the step was created doesn't mean that all the values passed above were set correctly.  We need to assert each one.

Reply:
I agree. I thought the goal was just to make sure these setters are called somewhere so that they are covered (in case we remove on of them accidentally). I was too pragmatic there 😄

Reply:
I'm almost done refactoring the test for this and will merge once I'm done.